pkg/cmd/roachprod-microbench/clean.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2024 The Cockroach Authors.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/cockroachdb/errors"
+)
+
+type cleanConfig struct {
+	inputFilePath  string
+	outputFilePath string
+}
+
+type clean struct {
+	cleanConfig
+	inputFile *os.File
+}
+
+func defaultCleanConfig() cleanConfig {
+	return cleanConfig{
+		inputFilePath:  "",
+		outputFilePath: "",
+	}
+}
+
+func newClean(config cleanConfig) (*clean, error) {
+	file, err := os.Open(config.inputFilePath)
+	if err != nil {
+		return nil, err
+	}
+
+	return &clean{cleanConfig: config, inputFile: file}, nil
+}
+
+func (c *clean) writeCleanOutputToFile(cleanedBenchmarkOutputLog benchmarkExtractionResult) error {
+
+	if err := os.MkdirAll(filepath.Dir(c.outputFilePath), os.ModePerm); err != nil {
```

Comment:
FWIW, I find this behaviour quite unusual for a command line utility: if I pass `foo/bar/baz/path.txt` to a command, I wouldn't expect it to create the entire directory tree to write to `path.txt`. It's generally better to have the caller be responsible for passing a valid path to begin with.

It makes handling these paths more explicit and also makes errors more evident, as writing to the path would fail if the path passed is invalid.

Reply:
@renatolabs Good point! Since this is meant to be added as a CI step, I was thinking in the way that we can create the directory in the code itself.